### PR TITLE
feat(runtime): support Kimi OpenAI protocol

### PR DESCRIPTION
**Pull request by Nyvo-io**
## Summary

- let the existing Kimi Coding Plan connection select Anthropic Messages or OpenAI Chat per model
- add Kimi OpenAI transport handling for reasoning dialect lifetime, explicit empty reasoning, tool replay, and usage capture
- preserve provider-owned reasoning metadata across RuntimeEvent and missing-ledger StoredMessage recovery
- keep unsupported or unmarked private thinking out of provider replay without dropping paired tool calls and results
- preserve recovered assistant step identity so reasoning, text, and tool calls reconstruct as one provider message

## Context

This is the flat Runtime protocol PR requested in the maintainer review on #1451. Request-trace evidence and the benchmark harness are intentionally excluded and will be submitted separately.

Refs #1269.

## Verification

- Biome format: 1,084 files passed
- Biome lint: 2,126 files passed
- all workspace typechecks passed
- `git diff --check` passed
- TDD regression reproduced the split recovered step before the fix and passed after it
- Kimi/Anthropic product protocol suite: 11 passed
- focused backend/backfill suites: 170 passed
- full `npm test`: all workspaces passed
- independent headless runner: 1,328 passed, 2 skipped
- local `review-agent` full base-diff review: no actionable correctness findings

No paid Kimi or Harbor benchmark was run; benchmark execution remains in the later benchmark PR.

### CODE REVIEW

File: packages/runtime/src/model-history.ts [hunk 1]
```diff
@@ -502,7 +485,14 @@ export function buildRuntimeEventModelReplayPlan(
...
-          signature: event.content.signature,
+          ...(event.content.signature ? { signature: event.content.signature } : {}),
+          ...(event.content.providerOptions !== undefined
+            ? {
+                providerOptions: event.content.providerOptions as NonNullable<
+                  ModelMessage['providerOptions']
+                >,
+              }
+            : {}),
           ...(event.refs?.providerEventId ? { stepId: event.refs.providerEventId } : {}),
```
Comment (Astro-Han) [inline packages/runtime/src/model-history.ts]: **[P2] Keep recovered reasoning in the tool-call step**

With a missing RuntimeEvent ledger, `backfillRuntimeEventsFromStoredMessages()` puts `storedMessageId` on the recovered assistant text and thinking events. The recovered tool call gets `stepId`. This line only reads `providerEventId`, so the reasoning has no step ID and is materialized as a separate assistant message instead of being grouped with its tool call.

I reproduced this through backfill, replay-plan construction, and materialization. Please use `providerEventId ?? storedMessageId` for assistant text and thinking, and add a regression test that combines empty reasoning, a tool call, missing-ledger recovery, and backend reconstruction.

**Review by Astro-Han — COMMENTED** [review]
One P2 recovery issue is noted inline. The rest of the change uses the existing Kimi provider and Runtime replay path.

File: packages/runtime/src/model-history.ts (hunk 1 above)
Comment (Nyvo-io) [reply packages/runtime/src/model-history.ts]: Fixed in c38ecf76 after rebasing onto current main. Replay-plan step identity now resolves assistant text and thinking with `providerEventId ?? storedMessageId`, while recovered tool calls continue to use their persisted `stepId`; this lets the materializer reconstruct all three parts as one assistant step. Added an end-to-end regression that combines non-empty assistant text, explicit empty Kimi reasoning, a tool call/result, missing-ledger backfill, backend recreation, and the actual OpenAI-compatible HTTP request. The test failed before the fix because the tool-call assistant had null content, then passed after the owner-layer change. Full workspace tests and the independent headless runner pass (1,328 passed, 2 skipped), and local review-agent found no actionable correctness issues.

**Review by Astro-Han — APPROVED** [review]
Reviewed `c38ecf7`. When ledger events are missing, the replay plan now uses `providerEventId ?? storedMessageId` as the assistant step ID.

The regression test follows the full path from backfill through backend reconstruction to the final OpenAI-compatible request. It covers assistant text, empty reasoning, the tool call, and its result.

All required checks pass. No findings. Approving.
